esmvaltool/diag_scripts/ocean3d/regions.py

```diff
@@ -0,0 +1,121 @@
+"""APPLICATE/TRR Ocean Diagnostics."""
+import logging
+import os
+import numpy as np
+from scipy.interpolate import interp1d
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def hofm_regions(region, lon2d, lat2d):
+    """Define regions for data selection.
+
+    Parameters
+    ----------
+    region: str
+        the name of the region
+    lon2d: 2d numpy array
+    lat2d: 2d numpy array
+
+    Returns
+    -------
+    indexesi: 1d numpy array
+        i indexes of the selected points
+    indexesj: 1d numpy array
+        j indexes of the selected points
+    """
+    if region == 'EB':
+        # Eurasian Basin of the Arctic Ocean
+        indi, indj = np.where((lon2d > 300) & (lat2d > 80))
+        indi2, indj2 = np.where((lon2d < 100) & (lat2d > 80))
+        indi3, indj3 = np.where((lon2d > 100) & (lon2d < 140) & (lat2d > 66))
+
+        indexesi = np.hstack((indi, indi2, indi3))
+        indexesj = np.hstack((indj, indj2, indj3))
+    elif region == 'AB':
+        # Amerasian basin of the Arctic Ocean
+        indi, indj = np.where((lon2d >= 260) & (lon2d <= 300) & (lat2d >= 80))
+        indi2, indj2 = np.where((lon2d >= 140) & (lon2d < 260) & (lat2d > 66))
+
+        indexesi = np.hstack((indi, indi2))
+        indexesj = np.hstack((indj, indj2))
+    elif region == 'Barents_sea':
+
+        indi, indj = np.where((lon2d >= 20) & (lon2d <= 55) & (lat2d >= 70)
+                              & (lat2d <= 80))
+
+        indexesi = indi
+        indexesj = indj
+    elif region == 'North_sea':
+        # Amerasian basin of the Arctic Ocean
+        indi, indj = np.where((lon2d >= 355) & (lon2d <= 360) & (lat2d >= 50)
+                              & (lat2d <= 60))
+        indi2, indj2 = np.where((lon2d >= 0) & (lon2d <= 10) & (lat2d >= 50)
+                                & (lat2d <= 60))
+
+        indexesi = np.hstack((indi, indi2))
+        indexesj = np.hstack((indj, indj2))
+    else:
+        print('Region {} is not recognized'.format(region))
+    return indexesi, indexesj
+
+
+def transect_points(transect, mult=2):
+    """Return a collection of points for transect.
```

Comment:
How does this compare with the transects in the ocean diagnostics and the `extract_transect` preprocessor?